largest-series-product/largest_series_product_test.go

```diff
@@ -56,6 +66,9 @@ const pe1k = "73167176531330624919225119674426574742355349194934" +
 	"71636269561882670428252483600823257530420752963450"
 
 func TestLargestSeriesProduct(t *testing.T) {
+	if TestVersion != testVersion {
+		t.Fatalf("Found TestVersion = %v, want %v", TestVersion, testVersion)
+	}
```

Comment:
I was suggesting that in line 71/84 below, the test could read `case int64(p) != test.product:`.  That would allow existing incorrect solutions to continue to compile.  They would pass tests on 64 bit architectures but fail on 32.  Without the conversion, we're forcing incorrect solutions to fail at compile time.  With the cast, lots of people will write incorrect code.  Others could nitpick these solutions then.  The advantage is that people will discuss and think about it more than if they just saw a compile error and fixed without stopping to think why it should be an int64.  I don't know, maybe that's hokey and we should write tests for correctness.  It's just an idea.